### PR TITLE
fix: samconfig design accuracy

### PR DESCRIPTION
**Pull request by sriram-mv**
Why is this change necessary?

* Implementation based on the samconfig design changed course based on
well reasoned discussions, but was not reflected back onto the design
doc.

How does it address the issue?

* Design reflects more of present day implementation of `samconfig.toml`

What side effects does this change have?

* None

TODO

* Add more detail on multiple configuration files and its future.

*Issue #, if available:*

*Description of changes:*

*Checklist:*

- [ ] Write Design Document ([Do I need to write a design document?](https://github.com/awslabs/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.rst#design-document))
- [ ] Write unit tests
- [ ] Write/update functional tests
- [ ] Write/update integration tests
- [ ] `make pr` passes
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: designs/sam-config.md [hunk 1]
```diff
@@ -38,53 +38,55 @@ Tenets
...
-The configuration file: `samconfig.toml` will come with a top level version key that specifies the version of the configuration file. This version can then be used to determine if a given configuration file works with a version of SAM CLI.
+The configuration file: `samconfig.toml` will come with a top level version key that specifies the version of the configuration file based on the spec of the file. This version can then be used to determine if a given configuration file works with a given version of SAM CLI.
 
 It also paves the forward when major changes need to be made to the configuration file and add a version bump to the config file version
 
 ```
 version = 0.1
 ```
 
+* SAM CLI expects the version format to follow a float format
```
Comment (sanathkr) [inline designs/sam-config.md]: you mean semantic version format? 0.1.1 is not a valid float

File: designs/sam-config.md [hunk 2]
```diff
@@ -265,7 +252,7 @@ Implementation
 CLI Changes
 -----------
 
-New command line argument is added per command called `--env` to be able to specify non default environment section within a config file.
+New command line argument is added per command called `--config-prefix` to be able to specify non default environment section within a config file.
```
Comment (sanathkr) [inline designs/sam-config.md]: Can you make a note somewhere that says --config-file prefix is not implemented yet?

File: designs/sam-config.md (hunk 1 above)
Comment (sriram-mv) [reply designs/sam-config.md]: Done

File: designs/sam-config.md (hunk 2 above)
Comment (sriram-mv) [reply designs/sam-config.md]: Done

**Pull request merged**